registry/repository_test.go

```diff
@@ -0,0 +1,468 @@
+	invalidDigest            = "invaliddigest"
+	algo                     = "sha256"
+	validDigestWithAlgo      = algo + ":" + validDigest
+	validDigestWithAlgo2     = algo + ":" + validDigest2
+	validDigestWithAlgo3     = algo + ":" + validDigest3
+	validDigestWithAlgo4     = algo + ":" + validDigest4
+	validDigestWithAlgo5     = algo + ":" + validDigest5
+	validHost                = "localhost"
+	validRegistry            = validHost + ":5000"
+	invalidHost              = "badhost"
+	invalidRegistry          = invalidHost + ":5000"
+	validRepo                = "test"
+	msg                      = "message"
+	errMsg                   = "error message"
+	mediaType                = "application/json"
+	validReference           = validRegistry + "/" + validRepo + "@" + validDigest
+	referenceWithInvalidHost = invalidRegistry + "/" + validRepo + "@" + validDigest
+	invalidReference         = "invalid reference"
+	orasManifestMediaType    = "application/vnd.cncf.oras.artifact.manifest.v1+json"
+	createdTag               = "io.cncf.oras.artifact.created"
+	joseTag                  = "application/jose+json"
+	validTimestamp           = "2022-07-29T02:23:10Z"
+	validPage                = `
+	{
+		"references": [{}],
+		"referrers": [
+			{
+				"artifactType": "application/vnd.cncf.notary.v2.signature",
+				"mediaType": "application/vnd.cncf.oras.artifact.manifest.invalid",
+				"digest": "localhost:5000/test@1834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f2"
+			},
+			{
+				"artifactType": "application/vnd.cncf.notary.v2.signature",
+				"mediaType": "application/vnd.cncf.oras.artifact.manifest.v1+json",
+				"digest": "localhost:5000/test@9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0"
+			}
+		]
+	}`
+	validManifest = `
+	{
+		"blobs": [
+			{
+				"mediaType": "application/jose+json",
+				"digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+				"size": 32654
+			}
+		],
+		"annotations": {
+			"io.cncf.notary.x509chain.thumbprint#S256": 
+			"[\"B7A69A70992AE4F9FF103EBE04A2C3BA6C777E439253CE36562E6E98375068C3\",\"932EB6F5598435D4EF23F97B0B5ACB515FAE2B8D8FAC046AB813DDC419DD5E89\"]"
+		}
+	}`
+	emptyManifest   = `{}`
+	invalidManifest = `}`
+)
+
+type args struct {
+	ctx             context.Context
+	reference       string
+	remoteClient    remote.Client
+	plainHttp       bool
+	digest          digest.Digest
+	annotations     map[string]string
+	subjectManifest notation.Descriptor
+	signature       []byte
+	fetchAll        func(ctx context.Context, fetcher content.Fetcher, desc ocispec.Descriptor) ([]byte, error)
+}
+
+type mockRemoteClient struct {
+}
+
+func (c mockRemoteClient) Do(req *http.Request) (*http.Response, error) {
+	switch req.URL.Path {
+	case "/v2/test/manifests/" + validDigest, "/v2/test/blobs/" + validDigestWithAlgo:
+		return &http.Response{
+			StatusCode: http.StatusOK,
+			Body:       ioutil.NopCloser(bytes.NewReader([]byte(msg))),
```

Comment:
nit: `ioutil ` will be depreciated(https://go.dev/doc/go1.16), we can use package `io` instead.

Reply:
good call, addressed.